src/tls/connection.c

```diff
@@ -373,8 +386,7 @@ request_dynamic_wsinstance (const char *fingerprint)
 
   /* send the fingerprint */
   debug (CONNECTION, "  -> success; sending fingerprint...");
-  assert (strlen (fingerprint) == FINGERPRINT_LENGTH);
-  if (!send_all (fd, fingerprint, FINGERPRINT_LENGTH, 5 * 1000000))
+  if (!send_all (fd, fingerprint->bytes, sizeof fingerprint->bytes, 5 * 1000000))
```

Comment:
As noted for the Fingerprint type definition already, I find it very confusing to use `->bytes` and `->str` from the same object. Here the intention is to send the ASCIIfied version, not the binary fingerprint, so this should use ->str?

Reply:
See my comments about this problem above.  Hopefully a comment will clear things up.

Do you have a better suggestion for the name `bytes`?

Reply:
Honestly, I'd drop the union and just use `sizeof fingerprint->str -1 /* exclude NUL terminator */`, or just do `strlen()` and let the compiler deal with it. Now that I understand the purpose, it's much clearer though -- documenting the purpose as a comment in the Fingerprint type declaration would also help. But TBH it still feels a bit like overkill.